src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -186,6 +191,13 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
     public class RankerWeight extends Weight {
         private final List<Weight> weights;
 
+        @Override
+        public boolean isCacheable(LeafReaderContext ctx) {
+            // Only cachable if model doesn't change?
+            // I *think* it should be the case it should be true
+            return true;
```

Comment:
I'd go with the BooleanWeight approach here: https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/search/BooleanWeight.java#L314

Reply:
So checked a bit more what's happening here:
- the valid way is to delegate to sub weights like what is done the BooleanWeight

But in fact it's just called to see if lucene can cache the docIds if this query was used as a filter.
So could very much just `throw new IllegalStateException("RankerWeight should not be used as a filter and cached");`